windows.c

```diff
@@ -0,0 +1,128 @@
+#include <stdlib.h>
+#include <windows.h>
+#include <Wininet.h>
+#include <ras.h>
+#include <tchar.h>
+#include <stdio.h>
+#include "common.h"
+
+void reportWindowsError(const char* action) {
+  LPTSTR pErrMsg = NULL;
+  DWORD errCode = GetLastError();
+  FormatMessage(FORMAT_MESSAGE_ALLOCATE_BUFFER|
+      FORMAT_MESSAGE_FROM_SYSTEM|
+      FORMAT_MESSAGE_ARGUMENT_ARRAY,
+      NULL,
+      errCode,
+      LANG_NEUTRAL,
+      pErrMsg,
+      0,
+      NULL);
+  printf("Error %s: %lu %s\n", action, errCode, pErrMsg);
+}
+
+// Figure out which Dial-Up or VPN connection is active; in a normal LAN connection, this should
+// return NULL. NOTE: For some reason this method fails when compiled in Debug mode but works
+// every time in Release mode.
+LPTSTR FindActiveConnection() {
+  DWORD dwCb = sizeof(RASCONN);
+  DWORD dwErr = ERROR_SUCCESS;
+  DWORD dwRetries = 5;
+  DWORD dwConnections = 0;
+  RASCONN* lpRasConn = NULL;
+  RASCONNSTATUS rasconnstatus;
+  rasconnstatus.dwSize = sizeof(RASCONNSTATUS);
+
+  //
+  // Loop through in case the information from RAS changes between calls.
+  //
+  while (dwRetries--) {
+    // If the memory is allocated, free it.
+    if (NULL != lpRasConn) {
+      HeapFree(GetProcessHeap(), 0, lpRasConn);
+      lpRasConn = NULL;
+    }
+
+    // Allocate the size needed for the RAS structure.
+    lpRasConn = (RASCONN*)HeapAlloc(GetProcessHeap(), 0, dwCb);
+    if (NULL == lpRasConn) {
+      dwErr = ERROR_NOT_ENOUGH_MEMORY;
+      break;
+    }
+
+    // Set the structure size for version checking purposes.
+    lpRasConn->dwSize = sizeof(RASCONN);
+
+    // Call the RAS API then exit the loop if we are successful or an unknown
+    // error occurs.
+    dwErr = RasEnumConnections(lpRasConn, &dwCb, &dwConnections);
+    if (ERROR_INSUFFICIENT_BUFFER != dwErr) {
+      break;
+    }
+  }
+  //
+  // In the success case, print the names of the connections.
+  //
+  if (ERROR_SUCCESS == dwErr) {
+    DWORD i;
+    for (i = 0; i < dwConnections; i++) {
+      RasGetConnectStatus(lpRasConn[i].hrasconn, &rasconnstatus);
+      if (rasconnstatus.rasconnstate == RASCS_Connected){
+        return lpRasConn[i].szEntryName;
+      }
+
+    }
+  }
+  return NULL; // Couldn't find an active dial-up/VPN connection; return NULL
+}
+
+int togglePac(bool turnOn, const char* pacUrl)
+{
+  int ret = RET_NO_ERROR;
+
+  INTERNET_PER_CONN_OPTION_LIST options;
+  DWORD   dwBufferSize = sizeof(options);
+  options.dwSize = dwBufferSize;
+  options.pszConnection = FindActiveConnection();
```

Comment:
The [documentation](https://msdn.microsoft.com/en-us/library/windows/desktop/aa385146%28v=vs.85%29.aspx) for INTERNET_PER_CONN_OPTION_LIST stats that for `pszConnection`

"Pointer to a string that contains the name of the RAS connection or NULL, which indicates the default or LAN connection, to set or query options on."

Have you tried just leaving it NULL to see if that works?  That would allow us to get rid of `FindActiveConnection()`.